packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -493,20 +493,8 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
      * Hook a middleware into the client
      * @param middleware to hook
      */
-    $use<T>(middleware: QueryMiddleware<T>)
-    $use<T>(namespace: 'all', cb: QueryMiddleware<T>) // TODO: 'all' actually means 'query', to be changed
-    $use<T>(namespace: 'engine', cb: EngineMiddleware<T>)
-    $use<T>(arg0: Namespace | QueryMiddleware<T>, arg1?: QueryMiddleware | EngineMiddleware<T>) {
-      // TODO use a mixin and move this into MiddlewareHandler
-      if (typeof arg0 === 'function') {
-        this._middlewares.query.use(arg0 as QueryMiddleware)
-      } else if (arg0 === 'all') {
-        this._middlewares.query.use(arg1 as QueryMiddleware)
-      } else if (arg0 === 'engine') {
-        this._middlewares.engine.use(arg1 as EngineMiddleware)
-      } else {
-        throw new Error(`Invalid middleware ${arg0}`)
-      }
+    $use<T>(middleware: QueryMiddleware<T>) {
+      this._middlewares.use(middleware as QueryMiddleware<unknown>)
```

Comment:
Quick read: just wondering if adding `<unknown>` changes something for our users?

Reply:
I don't think so, users use generated types. Will change it in a sec